android/src/main/java/com/rncamerakit/RNCameraKitPackage.kt

```diff
@@ -1,22 +1,44 @@
 package com.rncamerakit
 
-import com.facebook.react.ReactPackage
+import com.facebook.react.TurboReactPackage
 import com.facebook.react.bridge.NativeModule
 import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.module.annotations.ReactModuleList
+import com.facebook.react.module.model.ReactModuleInfo
+import com.facebook.react.module.model.ReactModuleInfoProvider
 import com.facebook.react.uimanager.ViewManager
 import java.util.*
 
-class RNCameraKitPackage : ReactPackage {
-    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
-        val modules: MutableList<NativeModule> = ArrayList()
-        val cameraModule = RNCameraKitModule(reactContext)
-        modules.add(cameraModule)
-        return modules
-    }
+@ReactModuleList(nativeModules = [RNCameraKitModule::class])
+class RNCameraKitPackage : TurboReactPackage() {
 
     override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
         val viewManagers: MutableList<ViewManager<*, *>> = ArrayList()
         viewManagers.add(CKCameraManager())
         return viewManagers
     }
+
+    override fun getModule(s: String, reactApplicationContext: ReactApplicationContext): NativeModule? {
+        when (s) {
+            RNCameraKitModule.REACT_CLASS -> return RNCameraKitModule(reactApplicationContext)
+        }
+        return null
+    }
+
+    override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
+        return ReactModuleInfoProvider {
+            val moduleInfos: MutableMap<String, ReactModuleInfo> = HashMap()
+            val isTurboModule = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
+            moduleInfos[RNCameraKitModule.REACT_CLASS] = ReactModuleInfo(
+                RNCameraKitModule.REACT_CLASS,
+                RNCameraKitModule.REACT_CLASS,
+                false,  // canOverrideExistingModule
+                false,  // needsEagerInit
+                true,  // hasConstants
+                false,  // isCxxModule
+                isTurboModule // isTurboModule
+            )
```

Comment:
Isn't this constructor deprecated? Which RN version are you targeting for the new arch?

Reply:
I checked on RN 0.75

Reply:
Changed in https://github.com/teslamotors/react-native-camera-kit/commit/b8417d8c5e2d767613a6f1b0baa2f5e8486715bd